### PR TITLE
Tutorial on evaluating expression prediction models

### PR DESCRIPTION
**Pull request by Hrovatin**
TODOs before merging:
 - [ ] Add metrics to pertpy main
 - [ ] Correct metric calls
 
Metrics to be added: 
- [ ] 2-way distances can be directly added to pertpy distances. To add mean_var_distn, ilisi (not so necessary as does not work well), MahalanobisDistance. In https://github.com/theislab/pertpy/blob/9c23791f3287f2265fb8e8bc7f0d3110fec977bf/pertpy/tools/_distances/_distances.py#L915
- [ ] 3-way distances - how do we approach this? A new class? Probably some of the preprocessing etc could be shared with the current Distances class - the main thing is to have the 3th group as input. These metrics are currently in different functions in https://github.com/theislab/pertpy/blob/hackathon-metrics/pertpy/tools/_metrics_3g.py

General metrics TODOs:
- [ ] All metrics should be further tested for edge cases and sparse/non-sparse inputs. 
- [ ] Some metrics would need to be inverted as they currently measure similarity not distance.
- [ ] Some metrics require a specific order of what is x,y,z input (control, perturbed, predicted perturbed) so that needs to be clear in the documentation.

During the hackathon other things were added, but not used in the tutorial: 
- Different aggregation functions for single-moment metrics: https://github.com/theislab/pertpy/blob/9c23791f3287f2265fb8e8bc7f0d3110fec977bf/pertpy/tools/_distances/_distances.py#L48
- Bootstrapping cells for metric computation: https://github.com/theislab/pertpy/blob/9c23791f3287f2265fb8e8bc7f0d3110fec977bf/pertpy/tools/_distances/_distances.py#L248

### CODE REVIEW

File: expression_prediction_evaluation.ipynb [hunk 1]
```diff
@@ -0,0 +1,469 @@
+{
```
Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <p>Great introduction!</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645446'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <blockquote>The lFCs can be modelled with Cauchy distibution.</blockquote><p>with a </p><blockquote>refer to pertpy documentation</blockquote><p>refer to <strong>the</strong></p><p> </p><blockquote>siulated</blockquote><p>typo</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645447'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <blockquote>as input control state</blockquote><p>as input a control </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645449'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <blockquote>Furthemore</blockquote><p>typo</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645450'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <blockquote>chalenging</blockquote><p>typo</p><p> </p><blockquote>assymetry</blockquote><p>typo</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645451'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.    from pertpy.tools._metrics_3g import compare_dist
</pre><p>This is not yet in pertpy, right? So we're waiting for Eljas to add it?</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645453'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.    from pertpy.tools import Distance
</pre><p>Antipattern of pertpy usage. We usually <code>import pertpy as pt</code> and then <code>pt.tl.BLA</code></p><p> </p><p> </p><p>But I guess this is also a relict of your fork. We need everything in main pertpy</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1500645456'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

**Comment by Hrovatin** [comment]
I think the grammar should be fixed now.

**Comment by Zethson** [comment]
Thanks! I'll check it out again soon

File: expression_prediction_evaluation.ipynb [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
```
Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <blockquote>computing R2 between predicted and ground truth data</blockquote><p>Kind of obvious to us what this means, but IMO this requires a link or a small explanation here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1517822324'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <p>Are you using small `l? for LFC? It's LFC not IFC. It's hard to read.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1517822326'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <p>I'd wrap the things like <code>sym_kldiv</code> in backticks to make them easier to read. Also in subsequent cells</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1517822327'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (Zethson) [inline expression_prediction_evaluation.ipynb:1]: <blockquote>We want to evaluate how the prediction performs in relation to both groups: control condition input into the model and the target perturbed condition.</blockquote><p>I find this sentence a bit hard to understand.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/theislab/pertpy-tutorials/pull/37/discussion/#comment-1517822328'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

**Comment by Hrovatin** [comment]
Updated the above

**Pull request merged**